app/api/queue.v2/infrastructure/ApplicationRedisClient.ts

```diff
@@ -0,0 +1,37 @@
+import { config } from 'api/config';
+import Redis from 'redis';
+
+let clientPromise: Promise<Redis.RedisClient> | null = null;
+
+export const ApplicationRedisClient = {
+  async getInstance(createClient: typeof Redis['createClient'] = Redis.createClient) {
+    if (clientPromise) {
+      return clientPromise;
+    }
+
+    clientPromise = new Promise((resolve, reject) => {
+      const client = createClient(`redis://${config.redis.host}:${config.redis.port}`);
+      client.on('ready', () => {
+        resolve(client);
+      });
+      client.on('error', reject);
+    });
+
+    return clientPromise;
+  },
+
+  async close() {
+    if (clientPromise) {
+      (await clientPromise).quit();
+      clientPromise = null;
+      return true;
+    }
+
+    return false;
+  },
+};
+
+// eslint-disable-next-line camelcase
+export const TESTING_clean = () => {
```

Comment:
I expect that method not to be used anywhere else that is not the testing for this file. So I chose this because it is obvious to the eye, but also will give you trouble with eslint when you try to use it :) Of course is arbitrary, and we can decide for something else.
To give context, the problem is that function should't be needed, but because there's no injection in the part of the application that is going to use that code, this is a global singleton, and global singletons are difficult to test.